engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/JobPreparationActor.scala

```diff
@@ -20,76 +22,83 @@ case class JobPreparationActor(executionData: WorkflowExecutionActorData,
                                serviceRegistryActor: ActorRef)
   extends Actor with WdlLookup with WorkflowLogging {
 
-  override val workflowDescriptor: EngineWorkflowDescriptor = executionData.workflowDescriptor
-  override val workflowId = workflowDescriptor.id
-  override val executionStore: ExecutionStore = executionData.executionStore
-  override val outputStore: OutputStore = executionData.outputStore
-  override val expressionLanguageFunctions = factory.expressionLanguageFunctions(
+  implicit val ec: ExecutionContext = context.dispatcher
+
+  override lazy val workflowDescriptor: EngineWorkflowDescriptor = executionData.workflowDescriptor
+  override lazy val workflowId = workflowDescriptor.id
+  override lazy val executionStore: ExecutionStore = executionData.executionStore
+  override lazy val outputStore: OutputStore = executionData.outputStore
+  override lazy val expressionLanguageFunctions = factory.expressionLanguageFunctions(
     workflowDescriptor.backendDescriptor, jobKey, initializationData)
 
   override def receive = {
     case Start =>
-      val response = resolveAndEvaluateInputs(jobKey, expressionLanguageFunctions) flatMap { prepareJobExecutionActor } match {
-        case Success(m) => m
-        case Failure(f) => BackendJobPreparationFailed(jobKey, f)
-      }
+      val response = resolveAndEvaluateInputs(jobKey, expressionLanguageFunctions) map { prepareJobExecutionActor } recover {
+        case f => BackendJobPreparationFailed(jobKey, f)
+      } andThen { case _ => self ! PoisonPill }
```

Comment:
I'll have to double check but I think there was still some work to do in the receive method. So I didn't want to abruptly stop it but rather make it stop the next time it's in a stable state